mcap_vendor/CMakeLists.txt

```diff
@@ -0,0 +1,87 @@
+cmake_minimum_required(VERSION 3.5)
+project(mcap_vendor)
+
+## Compile options
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 17)
+  set(CMAKE_CXX_STANDARD_REQUIRED ON)
+  set(CMAKE_CXX_EXTENSIONS OFF)
+endif()
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic -Werror)
+endif()
+
+## Dependencies
+find_package(ament_cmake REQUIRED)
+
+
+## Define vendor macro
+macro(build_mcap_vendor)
+  include(FetchContent)
+  fetchcontent_declare(mcap
+    GIT_REPOSITORY https://github.com/foxglove/mcap.git
+    GIT_TAG releases/cpp/v0.1.1
+  )
+  fetchcontent_makeavailable(mcap)
+
+  fetchcontent_declare(lz4
+    GIT_REPOSITORY https://github.com/lz4/lz4.git
+    GIT_TAG v1.9.3
+  )
+  fetchcontent_makeavailable(lz4)
+
+  fetchcontent_declare(zstd
+    GIT_REPOSITORY https://github.com/facebook/zstd.git
+    GIT_TAG v1.5.2
+  )
+  fetchcontent_makeavailable(zstd)
+
+  file(GLOB _zstd_srcs
+    ${zstd_SOURCE_DIR}/lib/common/*.c
+    ${zstd_SOURCE_DIR}/lib/compress/*.c
+    ${zstd_SOURCE_DIR}/lib/decompress/*.c
+    ${zstd_SOURCE_DIR}/lib/decompress/*.S
+  )
+  add_library(
+    mcap SHARED
+    src/main.cpp
+    ${_zstd_srcs}
+    ${lz4_SOURCE_DIR}/lib/lz4.c
+  )
+
+  set(_mcap_include_dir ${mcap_SOURCE_DIR}/cpp/mcap/include)
+
+  target_include_directories(mcap SYSTEM PUBLIC
+    "$<BUILD_INTERFACE:${_mcap_include_dir}>"
+    "$<INSTALL_INTERFACE:include>")
+
+  install(
+    DIRECTORY
+      ${_mcap_include_dir}/mcap
+    DESTINATION
+      ${CMAKE_INSTALL_PREFIX}/include
+  )
+
+  install(TARGETS mcap EXPORT export_mcap)
+
+  install(
+    EXPORT export_mcap
+    FILE mcap-config.cmake
+    DESTINATION share/cmake/mcap/
+    NAMESPACE "mcap_vendor::"
+    EXPORT_LINK_INTERFACE_LIBRARIES)
```

Comment:
I guess this block is unnecessary if you use `mcap_vendor` as namespace, because ament_export_targets do the s similar thing. https://github.com/ament/ament_cmake/blob/2bf27ce5ad01af340e1f4efd44232c1067d0dbda/ament_cmake_export_targets/cmake/ament_cmake_export_targets_package_hook.cmake#L27-L34

Reply:
I tried changing this to `mcap::`, but then `target_link_libraries(... mcap_vendor::mcap)` didn't work. What do you think is the best name to use?